pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessor.java

```diff
@@ -207,11 +207,7 @@ private boolean needProcessStarTrees() {
     List<StarTreeV2Metadata> starTreeMetadataList = _segmentMetadata.getStarTreeV2MetadataList();
     // There are existing star-trees, but if they match the builder configs exactly,
     // then there is no need to generate the star-trees
-    if (starTreeMetadataList != null && !StarTreeBuilderUtils
-        .shouldRemoveExistingStarTrees(starTreeBuilderConfigs, starTreeMetadataList)) {
-      return false;
-    }
-    return !starTreeBuilderConfigs.isEmpty();
+    return StarTreeBuilderUtils.existingStarTreesNeedChange(starTreeBuilderConfigs, starTreeMetadataList);
```

Comment:
@klsince Looks like the needProcessStarTrees was broken (caught by testStarTreeTriggering IT) wherein, this function returned false if existing startrees were to be removed. I've tries fixing it with these changes. Can you have a look?

Reply:
Good catch! 

Instead of changing the shouldRemoveExistingStarTrees(), could it work in this way? 
```
// falling out of the if-check means there are new changes:
// either adding ST trees per new configs, or remove all existing ST trees
return !starTreeBuilderConfigs.isEmpty() || starTreeMetadataList != null; 
```

Reply:
bump ^